data/contract_docs/contract.md

```diff
@@ -50,6 +50,12 @@ There are two named cases for when a response is lacking data and also doesn't h
 
 We retrieve one or more data items from a datasource but an item has no values, that item is said to be an "**_Empty value_**". In this case, the required dataframe fields should still be present (but the fields themselves each have no values).
 
+## Error Responses
+
+Errors are returned from outside the dataframes using the `Error` property of a [DataResponse](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend#DataResponse).
+
+When an error is returned with the DataResponse, a single frame with no fields may be included as well. If the error is present, this will not be considered "No Data". This frame is included so that metadata, in particular a Frame's `ExecutedQueryString`, can be returned to the caller with an error.
```

Comment:
For severity, will do a deprecation PR later as mentioned in the PR notes.

I didn't think of returning an error from QueryData. Returning an error here I assume means none of the queries can be successful. So perhaps this should be clarified to be only for errors that means no queries can be run due to a failure on the entire request (e.g. data source not found). WDYT?

I also just noticed https://github.com/grafana/grafana/pull/56832 - looking at that, I believe if an error is returned on QueryData, it would also drop the results from say other datasources that come in on the same HTTP request that (at least, that is what the `errgroup` `Wait()` will do I think)

Reply:
> For severity, will do a deprecation PR later as mentioned in the PR notes.

sorry, i must have skipped over the PR description 🤦 

> So perhaps this should be clarified to be only for errors that means no queries can be run due to a failure on the entire request (e.g. data source not found). WDYT?

yep, that's what i was thinking as well. i think @marefr and @wbrowne mentioned something like that during a recent meeting

Reply:
Yep that was at least my understanding too!

Reply:
Yep same, with the example `data source not found`, maybe a more common one in a plugin would be failing to connect to a downstream service due to being unavailable or wrongly configured etc. Think that is what you meant though?